spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -260,11 +291,24 @@ object DistriOptimizer {
         models.mapPartitions(modelIter => {
           val modelCache = modelIter.next()
           parameters.aggregrateGradientPartition()
-          parameters.gradientPartition.div(ev.fromType(finishedModelNum))
+          if (!optimMethod.isInstanceOf[EAMSGD[T]]) {
```

Comment:
I think we can put this logic in the following if(optimMethod.isInstanceOf[EAMSGD[T]]). It will make code cleaner.